docs/develop/writingtests.rst

```diff
@@ -0,0 +1,79 @@
+.. _writingtests:
+
+Writing tests
+=============
+
+Tests in MicroPython are written in the path ``py/tests``:
+
+.. code-block:: bash
+   
+   .
+    ├── basics
+    ├── cmdline
+    ├── cpydiff
+    ├── esp32
+    ├── extmod
+    ├── feature_check
+    ├── float
+    ├── import
+    ├── inlineasm
+    ├── internal_bench
+    ├── io
+    ├── jni
+    ├── micropython
+    ├── misc
+    ├── multi_bluetooth
+    ├── multi_net
+    ├── net_hosted
+    ├── net_inet
+    ├── perf_bench
+    ├── pyb
+    ├── pybnative
+    ├── qemu-arm
+    ├── README
+    ├── run-internalbench.py
+    ├── run-multitests.py
+    ├── run-natmodtests.py
+    ├── run-perfbench.py
+    ├── run-tests
+    ├── run-tests-exp.py
+    ├── run-tests-exp.sh
+    ├── stress
+    ├── thread
+    ├── unicode
+    ├── unix
+    └── wipy
+
+There are subfolders maintained to categorize most tests. Add a test by creating a new file in one of the
+existing folders or in a new folder.
+
+For example, add the following code in a file ``print.py`` in the unix subdirectory:
+
+.. code-block:: python
+   
+   def print_one():
+    print(1)
+   
+   print_one()
+
+If you run your tests, this test should appear in the test output:
+
+.. code-block:: bash
+   
+   $ cd ports/unix
+   $ make tests
```

Comment:
The other way to run tests (which might be worth mentioning here) is:
```bash
$ cd tests
$ ./run-tests
```
Then to run on a board:
```bash
$ ./run-tests --target minimal --device /dev/ttyACM0
```
And to run only a certain set of tests (eg a directory):
```bash
$ ./run-tests -d basics
```